### ISSUE TITLE
Jose dependency

### ISSUE THREAD

**Issue by happysalada** [ydlr/mix2nix#4]
the jose dependency needs to be built with buildMix when building an elixir Project.

As an example the guardian elixir library depends on jose and is trying to load a struct. Building jose with rebar will not provide the output struct.

Here is my deps.nix
```
{ lib, beamPackages, overrides ? (x: y: { }) }:

let
  buildRebar3 = lib.makeOverridable beamPackages.buildRebar3;
  buildMix = lib.makeOverridable beamPackages.buildMix;
  buildErlangMk = lib.makeOverridable beamPackages.buildErlangMk;

  self = packages // (overrides self packages);

  packages = with beamPackages; with self; {
    gettext = buildMix rec {
      name = "gettext";
      version = "0.18.2";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1igmn69xzj5wpkblg3k9v7wa2fjc2j0cncwx0grk1pag7nqkgxgr";
      };

      beamDeps = [ ];
    };

    phoenix_live_view = buildMix rec {
      name = "phoenix_live_view";
      version = "0.15.7";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0vi7agzrmlpr0jsbmaasm6sn6892wp60ifdky78740j2cbv6qx9s";
      };

      beamDeps = [ jason phoenix phoenix_html telemetry ];
    };

    phoenix_pubsub = buildMix rec {
      name = "phoenix_pubsub";
      version = "2.0.0";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0wgpa19l6xar0k2m117iz2kq3cw433llp07sqswpf5969y698bf5";
      };

      beamDeps = [ ];
    };

    proquint = buildMix rec {
      name = "proquint";
      version = "1.0.2";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1wwgqz60pwddxbw780wlcn3sjg5ghjhh8lpm74dj1rk7x8wgviib";
      };

      beamDeps = [ ];
    };

    elixir_make = buildMix rec {
      name = "elixir_make";
      version = "0.6.2";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1na8agkks1hrwq1lxfj4yd96bvfcs4hk7mbra9z6lli2vanrxr03";
      };

      beamDeps = [ ];
    };

    stripity_stripe = buildMix rec {
      name = "stripity_stripe";
      version = "2.10.0";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1z1bzs3an3d8mpxf0gc76vj2r29l427ia1j1ijzm85flnsd3zfim";
      };

      beamDeps = [ hackney jason uri_query ];
    };

    remote_ip = buildMix rec {
      name = "remote_ip";
      version = "0.2.1";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1q78grmmscb4n4qkc1i3754vij25iyh1dq9pkxqxag6cc6jb2yif";
      };

      beamDeps = [ combine inet_cidr plug ];
    };

    telemetry = buildRebar3 rec {
      name = "telemetry";
      version = "0.4.3";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0hc0fr2bh97wah9ycpm7hb5jdqr5hnl1s3b2ibbbx9gxbwvbhwpb";
      };

      beamDeps = [ ];
    };

    dialyxir = buildMix rec {
      name = "dialyxir";
      version = "1.1.0";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "16hbdhkbqq69v452vx3c0d26mmadkmwr6nvdpr72c5azqi4qxsh7";
      };

      beamDeps = [ erlex ];
    };

    comeonin = buildMix rec {
      name = "comeonin";
      version = "5.3.2";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "012zr4s7b5bipng6yszqxkqr1lcv7imf8gyvxad56jachh1396fh";
      };

      beamDeps = [ ];
    };

    bcrypt_elixir = buildMix rec {
      name = "bcrypt_elixir";
      version = "2.3.0";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0s9psinz913l690xbcrl21m23zwinw4r2ypjgg7ybl3f9wfxd09c";
      };

      beamDeps = [ comeonin elixir_make ];
    };

    absinthe_plug = buildMix rec {
      name = "absinthe_plug";
      version = "1.5.8";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0nkfk5gqbg8yvlysgxfcak7y4lsy8q2jfyqyhql5hwvvciv43c5v";
      };

      beamDeps = [ absinthe plug ];
    };

    exprintf = buildMix rec {
      name = "exprintf";
      version = "0.2.1";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0a6s05hfs1hymb40rkn72lwn836nqlrjbj7wfxmfb45yh34fi810";
      };

      beamDeps = [ ];
    };

    swoosh = buildMix rec {
      name = "swoosh";
      version = "0.25.6";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0azmsnam3zqagi3ydph4a0rnm6ckwfrpvlnl6wmq23g5hgqzi5sx";
      };

      beamDeps = [ cowboy hackney jason mime plug_cowboy ];
    };

    inet_cidr = buildMix rec {
      name = "inet_cidr";
      version = "1.0.4";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1g61i08cizr99ivy050lv8fmvnwia9zmipfvlwff8jkhi40x78k4";
      };

      beamDeps = [ ];
    };

    phoenix_ecto = buildMix rec {
      name = "phoenix_ecto";
      version = "4.2.1";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1h23fy3pylaszh3l2zafq1a7fjwlwb3yw7dy09p0mb4wi6p1p2j7";
      };

      beamDeps = [ ecto phoenix_html plug ];
    };

    jose = buildMix rec {
      name = "jose";
      version = "1.11.1";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1i8szzpmiqc7xdv0lp38ng9fild7c5182b4pzkx4qbydnfgnr3q7";
      };

      beamDeps = [ ];
    };

    makeup_elixir = buildMix rec {
      name = "makeup_elixir";
      version = "0.15.1";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1j3fdjp915nq44c55gwysscryyqivjnaaign0wman1sb4drw2s6v";
      };

      beamDeps = [ makeup nimble_parsec ];
    };

    ssl_verify_fun = buildRebar3 rec {
      name = "ssl_verify_fun";
      version = "1.1.6";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1026l1z1jh25z8bfrhaw0ryk5gprhrpnirq877zqhg253x3x5c5x";
      };

      beamDeps = [ ];
    };

    jason = buildMix rec {
      name = "jason";
      version = "1.2.2";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0y91s7q8zlfqd037c1mhqdhrvrf60l4ax7lzya1y33h5y3sji8hq";
      };

      beamDeps = [ decimal ];
    };

    ecto = buildMix rec {
      name = "ecto";
      version = "3.6.1";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0lv5mypffqaxqkzhv78d9qvhdy4ckxs8lj150ygv2iq4k552kcyb";
      };

      beamDeps = [ decimal jason telemetry ];
    };

    telemetry_metrics = buildMix rec {
      name = "telemetry_metrics";
      version = "0.6.0";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0gawj416w0fpc6vpz3c6xbbaw7z9rqhl791aa4skh92b8snngzjw";
      };

      beamDeps = [ telemetry ];
    };

    ex_doc = buildMix rec {
      name = "ex_doc";
      version = "0.24.2";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1nmpdxydbc1khcayab98gfv7km2qrqmgp1s64kjdkf11x3cy2d71";
      };

      beamDeps = [ earmark_parser makeup_elixir makeup_erlang ];
    };

    plug_crypto = buildMix rec {
      name = "plug_crypto";
      version = "1.2.2";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1nxnxj62iv4yvm4771jbxpj3l4brn2crz053y12s998lv5x1qqw7";
      };

      beamDeps = [ ];
    };

    mimerl = buildRebar3 rec {
      name = "mimerl";
      version = "1.2.0";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "08wkw73dy449n68ssrkz57gikfzqk3vfnf264s31jn5aa1b5hy7j";
      };

      beamDeps = [ ];
    };

    excoveralls = buildMix rec {
      name = "excoveralls";
      version = "0.14.1";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1c68i4v3vfq4qdrnprmyv3apcbwvhi6yml7kq4619p7rijgqyn2a";
      };

      beamDeps = [ hackney jason ];
    };

    makeup_erlang = buildMix rec {
      name = "makeup_erlang";
      version = "0.1.1";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1fvw0zr7vqd94vlj62xbqh0yrih1f7wwnmlj62rz0klax44hhk8p";
      };

      beamDeps = [ makeup ];
    };

    observer_cli = buildRebar3 rec {
      name = "observer_cli";
      version = "1.6.2";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "07rl96r4hhqldzy73a13bxy2drmrzyd0j2ibqjnlks50rkjbjgf2";
      };

      beamDeps = [ recon ];
    };

    phoenix_html = buildMix rec {
      name = "phoenix_html";
      version = "2.14.3";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "03z8r285znlg25yi47d4l59s7jq58y4dnhvbxgp16npkzykrgmpg";
      };

      beamDeps = [ plug ];
    };

    file_system = buildMix rec {
      name = "file_system";
      version = "0.2.10";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1p0myxmnjjds8bbg69dd6fvhk8q3n7lb78zd4qvmjajnzgdmw6a1";
      };

      beamDeps = [ ];
    };

    phoenix_live_reload = buildMix rec {
      name = "phoenix_live_reload";
      version = "1.3.1";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1vx5flhkycn6w0gv6sg4b51b7vxwkzi9kccbpjr1qnmqmfsjdbpk";
      };

      beamDeps = [ file_system phoenix ];
    };

    metrics = buildRebar3 rec {
      name = "metrics";
      version = "1.0.1";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "05lz15piphyhvvm3d1ldjyw0zsrvz50d2m5f2q3s8x2gvkfrmc39";
      };

      beamDeps = [ ];
    };

    nimble_parsec = buildMix rec {
      name = "nimble_parsec";
      version = "1.1.0";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0jzmr0x3s6z16c74gvrk9aqc10brn6a1dwa8ywzr2vkhdgb35sq8";
      };

      beamDeps = [ ];
    };

    idna = buildRebar3 rec {
      name = "idna";
      version = "6.1.1";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1sjcjibl34sprpf1dgdmzfww24xlyy34lpj7mhcys4j4i6vnwdwj";
      };

      beamDeps = [ unicode_util_compat ];
    };

    number = buildMix rec {
      name = "number";
      version = "1.0.3";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "060gr8r24mygkgvdwc0i8bvrwyygr5n14c54lrjsjb3b16y7nffx";
      };

      beamDeps = [ decimal ];
    };

    hackney = buildRebar3 rec {
      name = "hackney";
      version = "1.17.4";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "05kbk3rpw2j3cb9pybikydxmi2nm5pidpx0jsm48av2mjr4zy5ny";
      };

      beamDeps = [ certifi idna metrics mimerl parse_trans ssl_verify_fun unicode_util_compat ];
    };

    httpoison = buildMix rec {
      name = "httpoison";
      version = "1.8.0";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0fiwkdrbj7mmz449skp7laz2jdwsqn3svddncmicd46gk2m9w218";
      };

      beamDeps = [ hackney ];
    };

    plug_cowboy = buildMix rec {
      name = "plug_cowboy";
      version = "2.5.0";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "06p7rmx01fknkf0frvjjaqs3qsz6066aa41qyd378n72lljqjb2v";
      };

      beamDeps = [ cowboy cowboy_telemetry plug telemetry ];
    };

    guardian = buildMix rec {
      name = "guardian";
      version = "2.1.1";
      # enableDebugInfo = true;
      # postConfigure = "echo $ERL_LIBS";
      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1y7hifpl34cvacwmp7r2izklmz3gp6c804r9l1mhvd76gjx8g6qq";
      };

      beamDeps = [ jose plug ];
    };

    postgrex = buildMix rec {
      name = "postgrex";
      version = "0.15.9";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1vmd63vxwz8knid424b0rbp200vj7q7rz3xp98yj5cjc7q81j1v1";
      };

      beamDeps = [ connection db_connection decimal jason ];
    };

    timex = buildMix rec {
      name = "timex";
      version = "3.7.5";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1r3l50p8f8mxgghh079v1y5g02kzqr15ijbi7mkfzwl0lvf0hmm1";
      };

      beamDeps = [ combine gettext tzdata ];
    };

    cowboy = buildErlangMk rec {
      name = "cowboy";
      version = "2.9.0";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1phv0a1zbgk7imfgcm0dlacm7hbjcdygb0pqmx4s26jf9f9rywic";
      };

      beamDeps = [ cowlib ranch ];
    };

    parse_trans = buildRebar3 rec {
      name = "parse_trans";
      version = "3.3.1";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "12w8ai6b5s6b4hnvkav7hwxd846zdd74r32f84nkcmjzi1vrbk87";
      };

      beamDeps = [ ];
    };

    cowlib = buildRebar3 rec {
      name = "cowlib";
      version = "2.11.0";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1ac6pj3x4vdbsa8hvmbzpdfc4k0v1p102jbd39snai8wnah9sgib";
      };

      beamDeps = [ ];
    };

    exprof = buildMix rec {
      name = "exprof";
      version = "0.2.4";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0mv9218jbwrn5v1idcfknrywqd4hp75nl5a9sxsiqhpwdavbr108";
      };

      beamDeps = [ exprintf ];
    };

    ecto_sql = buildMix rec {
      name = "ecto_sql";
      version = "3.6.1";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0y871vj9s54dmg8snqhdcml6p7ccmd8n67mxzszvcy2r5lzmrwv6";
      };

      beamDeps = [ db_connection ecto postgrex telemetry ];
    };

    decimal = buildMix rec {
      name = "decimal";
      version = "2.0.0";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0xzm8hfhn8q02rmg8cpgs68n5jz61wvqg7bxww9i1a6yanf6wril";
      };

      beamDeps = [ ];
    };

    telemetry_poller = buildRebar3 rec {
      name = "telemetry_poller";
      version = "0.5.1";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1m1zcq65yz0wp1wx7mcy2iq37cyizbzrmv0c11x6xg0hj8375asc";
      };

      beamDeps = [ telemetry ];
    };

    combine = buildMix rec {
      name = "combine";
      version = "0.10.0";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "06s5y8b0snr1s5ax9v3s7rc6c8xf5vj6878d1mc7cc07j0bvq78v";
      };

      beamDeps = [ ];
    };

    recon = buildRebar3 rec {
      name = "recon";
      version = "2.5.1";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0j26nin8h3zzypppkdxsjiwgjc8jm8n73b6cikvdh8h1snvcc8ap";
      };

      beamDeps = [ ];
    };

    mime = buildMix rec {
      name = "mime";
      version = "1.6.0";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "19qrpnmaf3w8bblvkv6z5g82hzd10rhc7bqxvqyi88c37xhsi89i";
      };

      beamDeps = [ ];
    };

    db_connection = buildMix rec {
      name = "db_connection";
      version = "2.4.0";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1j6psw0dxq1175b6zcqpm6vavv4n6sv72ji57l8b6qczmlhnqhdd";
      };

      beamDeps = [ connection telemetry ];
    };

    cors_plug = buildMix rec {
      name = "cors_plug";
      version = "2.0.3";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1sls8rns2k48qrga0ngysbn9aknapmn3xfn28by1gqbcir0y2jpf";
      };

      beamDeps = [ plug ];
    };

    eflame = buildMix rec {
      name = "eflame";
      version = "1.0.1";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0kphkjlxg57r6xw1h0mrd6dsw49lix40ic70kl91743glra8ic70";
      };

      beamDeps = [ ];
    };

    erlex = buildMix rec {
      name = "erlex";
      version = "0.2.6";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0x8c1j62y748ldvlh46sxzv5514rpzm809vxn594vd7y25by5lif";
      };

      beamDeps = [ ];
    };

    sobelow = buildMix rec {
      name = "sobelow";
      version = "0.11.1";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0g558rki9198a2plgdadsbjxaalyh6nhma84jf0g95pzgqx3d5wq";
      };

      beamDeps = [ jason ];
    };

    ecto_psql_extras = buildMix rec {
      name = "ecto_psql_extras";
      version = "0.6.5";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0asqfr2yvrzwhi2f0wmwlz8r3mhmgnhh28jij8hrs08hh6kjdgjx";
      };

      beamDeps = [ ecto_sql postgrex table_rex ];
    };

    unicode_util_compat = buildRebar3 rec {
      name = "unicode_util_compat";
      version = "0.7.0";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "08952lw8cjdw8w171lv8wqbrxc4rcmb3jhkrdb7n06gngpbfdvi5";
      };

      beamDeps = [ ];
    };

    phoenix_live_dashboard = buildMix rec {
      name = "phoenix_live_dashboard";
      version = "0.4.0";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0mqfypsgi9678y6c73lrjrpvdnclmc08j3fcjx29x3hqb2g18lld";
      };

      beamDeps = [ ecto_psql_extras phoenix_html phoenix_live_view telemetry_metrics ];
    };

    nimble_csv = buildMix rec {
      name = "nimble_csv";
      version = "1.1.0";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0gwsjjfprhdkd3xvwymlh8y2vn691zddxrlv8an2r0q2qddpb1p9";
      };

      beamDeps = [ ];
    };

    makeup = buildMix rec {
      name = "makeup";
      version = "1.0.5";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1a9cp9zp85yfybhdxapi9haa1yykzq91bw8abmk0qp1z5p05i8fg";
      };

      beamDeps = [ nimble_parsec ];
    };

    connection = buildMix rec {
      name = "connection";
      version = "1.1.0";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1746n8ba11amp1xhwzp38yfii2h051za8ndxlwdykyqqljq1wb3j";
      };

      beamDeps = [ ];
    };

    cloudex = buildMix rec {
      name = "cloudex";
      version = "1.4.1";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0c98ygga0cv2ii94vnbgdb7fiyzla4vpqriafy0vw6vcdrvdvxyf";
      };

      beamDeps = [ httpoison jason timex tzdata ];
    };

    bunt = buildMix rec {
      name = "bunt";
      version = "0.2.0";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0lw3v9kwbbcy1v6ygziiky887gffwwmxvyg4r1v0zm71kzhcgxbs";
      };

      beamDeps = [ ];
    };

    table_rex = buildMix rec {
      name = "table_rex";
      version = "3.1.1";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "141404hwnwnpspvhs112j2la8dfnvkwr0xy14ff42w6nljmj72k7";
      };

      beamDeps = [ ];
    };

    plug = buildMix rec {
      name = "plug";
      version = "1.11.1";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "07w83cx4xx90x4l1kmil4lpby55lpw83jfw3y08pqn5vxx7lwli3";
      };

      beamDeps = [ mime plug_crypto telemetry ];
    };

    earmark_parser = buildMix rec {
      name = "earmark_parser";
      version = "1.4.13";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1fhlh9bnph5nqdhy7w69xzb7lra1b3v16mk4yb947bx0ydmc40nn";
      };

      beamDeps = [ ];
    };

    quantum = buildMix rec {
      name = "quantum";
      version = "3.3.0";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0rrgny21qcmh6grg0khwnxk7v11yrssihd017dw7x25kg89yz0rv";
      };

      beamDeps = [ crontab gen_stage telemetry ];
    };

    phoenix_swoosh = buildMix rec {
      name = "phoenix_swoosh";
      version = "0.2.0";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "07vbcxhyf0yycp434qvhw0sv2yhd5qpf1br997051j61nfkvzwgb";
      };

      beamDeps = [ phoenix phoenix_html swoosh ];
    };

    uri_query = buildMix rec {
      name = "uri_query";
      version = "0.1.2";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "037aw9m2w2kc5pmprzyzxbjiriw9kcijz6wb94v2ql4qdj0q3g73";
      };

      beamDeps = [ ];
    };

    cowboy_telemetry = buildRebar3 rec {
      name = "cowboy_telemetry";
      version = "0.3.1";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1bzhcdq12p837cii2jgvzjyrffiwgm5bsb1pra2an3hkcqrzsvis";
      };

      beamDeps = [ cowboy telemetry ];
    };

    certifi = buildRebar3 rec {
      name = "certifi";
      version = "2.6.1";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0zmvagzisnk7lj5pfipl19mjq9wn70i339hpbkfljf0vk6s9fk2j";
      };

      beamDeps = [ ];
    };

    ranch = buildRebar3 rec {
      name = "ranch";
      version = "1.8.0";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1rfz5ld54pkd2w25jadyznia2vb7aw9bclck21fizargd39wzys9";
      };

      beamDeps = [ ];
    };

    absinthe = buildMix rec {
      name = "absinthe";
      version = "1.6.4";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1bj1rpl4j9y47yi2h6pnw01nnry2s5idny4wkb5h9iw6dprir73f";
      };

      beamDeps = [ decimal nimble_parsec telemetry ];
    };

    phoenix = buildMix rec {
      name = "phoenix";
      version = "1.5.9";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0za2fpplmdq3axwng8736h6vz9wzlifhxy8apcgjy0bwlqhcwjvy";
      };

      beamDeps = [ jason phoenix_html phoenix_pubsub plug plug_cowboy plug_crypto telemetry ];
    };

    tzdata = buildMix rec {
      name = "tzdata";
      version = "1.1.0";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0d20s66kr7yqjrsp0b7ym85v8aqivk90d28frxdxrls8kdrm7x0q";
      };

      beamDeps = [ hackney ];
    };

    credo = buildMix rec {
      name = "credo";
      version = "1.5.5";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1pj7h9fk3i3rhxs65z8d64nwbnddhqj0dwy9bn2nm5cif2mj71nx";
      };

      beamDeps = [ bunt file_system jason ];
    };

    sentry = buildMix rec {
      name = "sentry";
      version = "8.0.5";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0z2bpm4j0x5liyw3d4575x7pc7s2vy54qs9ygdnqhbpmvfgq6wj9";
      };

      beamDeps = [ hackney jason plug plug_cowboy ];
    };

    mutex = buildMix rec {
      name = "mutex";
      version = "1.3.1";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "0nfyzwvz3a3lvsaxs9pvb5qg6a1nb6966m83f3m0la8zr50sdvv0";
      };

      beamDeps = [ ];
    };

    gen_stage = buildMix rec {
      name = "gen_stage";
      version = "1.1.0";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "01ck7g58ll6shc83v7y9r5ggcfj2xi0bacq786xg4zigs2k3cavz";
      };

      beamDeps = [ ];
    };

    crontab = buildMix rec {
      name = "crontab";
      version = "1.1.10";

      src = fetchHex {
        pkg = "${name}";
        version = "${version}";
        sha256 = "1186lif7k8i8paqcp2k8ppxlpqsr8f4v8xh8k6bskvd0s64xhiqk";
      };

      beamDeps = [ ecto ];
    };
  };
in
self
```

jose defaults to being built by buildRebar3, you can try to replace in my copy pasta and it will not successfully build anymore.
Let me know if you want more details on this.
It looks like it's another exception that will extend that case clause.

**Comment by happysalada** [comment]
Also regarding the default builder, It's just my opinion, but I think it should be buildMix.
Here is my rationale.
- Trying to consider adding git support. The git dependency format inside mix.lock is very sparse and doesn't include any information. However most mix project are probably going to try to use mix buildable dependencies. (I could be biased here as an elixir user, however I see erlang users using rebar2nix more).
- I think but cannot prove that the number of packages that don't work with buildMix is smaller than the ones that don't work with buildRebar3.

Just an opinion, feel free to ignore.

**Comment by ydlr** [comment]
It looks like we might be able to catch more cases if we take into account the builder used by a dependency's parent. If a package uses mix, then its dependencies should prefer mix. Same with rebar.

I'm also still thinking about using either buildHex or a custom abstraction to make it easier to override the builder for individual packages. It would be nice to simply be able to do:
```
overrides = (self: super: {
  jose = super.jose.override { builder = buildMix; };
}
```

I will try to find time to play with this in the next week or two.

**Comment by happysalada** [comment]
I like your idea about the parent builder.
However the ssl_fun error we had was a rebar dependency and the parent was a mix dependency (If I remember correctly).

It's a good idea to enable overrides. However I'm not sure many people would use it, the amound of people willing to actually mess with their `deps.nix` might not be that big. I feel it requires a bit of knowledge around the dependency process and knowledge of this potential fix. I might be wrong on that!

Thanks a lot for taking a look at this!

**Comment by happysalada** [comment]
Looking briefly at the mix repo, it seems the way they do it is that they check if there is a rebar.config file or a mix.lock in the main repo. If the dependency contains a mix.lock it is buildMix, if it contains a rebar.config it is buildRebar3
Perhaps we can do the same?

The tradeoff here is that it requires not only fetching the hash, but also the files themselves and checking them. It's more time-consuming but it could be more precise.

**Comment by ydlr** [comment]
Resolved in 0296e35a153d2692508188c4b0fffd0cc7fe1eb7
